src/Dialogue.js

```diff
@@ -26,42 +35,77 @@ export default class Dialogue {
     if (this.isComplete()) {
       return this.completeText;
     } else if (this.isBattle()) {
-      return `${this.conversation[this.progress].description}\n${this.startBattle()}`;
+      return `${this.activeText.description}\n${this.startBattle()}`;
     }
-    return this.conversation[this.progress].description;
+    return this.activeText.description;
   }
 
   /**
    * Restart the dialogue!
    */
   restart() {
-    this.progress = 0;
+    for (var text in this.conversation) {
+      this.activeText = text;
+      break;
+    }
```

Comment:
or maybe we also add a property called like `startKey` or something. then for here you would only need to do `this.activeText = this.conversation[this.startKey]`

we would need `startKey` and `progress` to be separate variables - `progress` is what's saved so that exiting/entering an area puts you in the same spot, and killing/restarting the game puts you in the same spot (which I think might be broken right now)